### PR TITLE
[OSS-ONLY] Remove extra check for helper functions for sp_describe_undeclared_parameters_internal

### PR DESCRIPTION
**Pull request by ahmed-shameem**
### Description
This commit removes the extra check (if condition) for helper functions for sp_describe_undeclared_parameters_internal which was introduced in this [commit](https://github.com/babelfish-for-postgresql/babelfish_extensions/commit/eda23eac1e40e8d6fd4453eb2d0d59c833008373). 

### Issues Resolved

BABEL-4869
Signed-off-by: Shameem Ahmed [shmeeh@amazon.com](mailto:shmeeh@amazon.com)

### Test Scenarios Covered ###
* **Use case based -**

* **Boundary conditions -**

* **Arbitrary inputs -**

* **Negative test cases -**

* **Minor version upgrade tests -**

* **Major version upgrade tests -**

* **Performance tests -**

* **Tooling impact -**

* **Client tests -**

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/src/procedures.c [hunk 1]
```diff
@@ -859,36 +814,20 @@ handle_where_clause_restargets_right(ParseState *pstate, Node *w_clause, List *e
...
-						break;
 					}
 					ref = (ColumnRef *) xpr->rexpr;
 					field = linitial(ref->fields);
 					res = (ResTarget *) palloc(sizeof(ResTarget));
 					res->type = ref->type;
 					res->name = field->val.str;
 					res->indirection = NIL; /* Unused for now */
-					res->val = (Node *) ref; /* Store the ColumnRef here if needed */
+					res->val = (Node *) ref;	/* Store the ColumnRef
```
Comment (timchang514) [inline contrib/babelfishpg_tsql/src/procedures.c]: nit: unnecessary diff

File: contrib/babelfishpg_tsql/src/procedures.c [hunk 2]
```diff
@@ -859,36 +814,20 @@ handle_where_clause_restargets_right(ParseState *pstate, Node *w_clause, List *e
 
 					if (nodeTag(xpr->rexpr) != T_ColumnRef)
 					{
-						xpr = (A_Expr *) arg;
-
-						if (nodeTag(xpr->rexpr) != T_ColumnRef)
+						if (is_sp_describe_undeclared_parameters)
```
Comment (timchang514) [inline contrib/babelfishpg_tsql/src/procedures.c:816]: does this need to be two separate if statements?

File: contrib/babelfishpg_tsql/src/procedures.c [hunk 3]
```diff
@@ -777,16 +731,17 @@ handle_where_clause_restargets_left(ParseState *pstate, Node *w_clause, List *ex
 					if (attrno == InvalidAttrNumber)
 					{
 						ereport(ERROR,
-						(errcode(ERRCODE_UNDEFINED_COLUMN),
-						 errmsg("column \"%s\" of relation \"%s\" does not exist",
-								name,
-								RelationGetRelationName(pstate->p_target_relation))));
+								(errcode(ERRCODE_UNDEFINED_COLUMN),
```
Comment (timchang514) [inline contrib/babelfishpg_tsql/src/procedures.c]: nit: unnecessary diff. 

Also, is this case different than the others? ie throwing an error rather than returning null result set?

File: contrib/babelfishpg_tsql/src/procedures.c [hunk 4]
```diff
@@ -737,38 +718,11 @@ handle_where_clause_restargets_left(ParseState *pstate, Node *w_clause, List *ex
...
-								is_supported_case_sp_describe_undeclared_parameters = false;
-								return extra_restargets;
-							}
-						}
-						ref = (ColumnRef *) xpr->lexpr;
-						field = linitial(ref->fields);
-						name = field->val.str;
-						attrno = attnameAttNum(pstate->p_target_relation, name, false);
-						if (attrno == InvalidAttrNumber)
+						if (is_sp_describe_undeclared_parameters)
```
Comment (timchang514) [inline contrib/babelfishpg_tsql/src/procedures.c:721]: same as below, does it need to be two if statements?

File: contrib/babelfishpg_tsql/src/procedures.c [hunk 5]
```diff
@@ -649,10 +630,10 @@ handle_where_clause_attnums(ParseState *pstate, Node *w_clause, List *target_att
 					if (attrno == InvalidAttrNumber)
 					{
 						ereport(ERROR,
-						(errcode(ERRCODE_UNDEFINED_COLUMN),
-						 errmsg("column \"%s\" of relation \"%s\" does not exist",
-								name,
-								RelationGetRelationName(pstate->p_target_relation))));
+								(errcode(ERRCODE_UNDEFINED_COLUMN),
```
Comment (timchang514) [inline contrib/babelfishpg_tsql/src/procedures.c]: nit: unnecessary diff.

Also, is this case different than the others? ie throwing an error rather than returning null result set?

File: contrib/babelfishpg_tsql/src/procedures.c (hunk 1 above)
Comment (ahmed-shameem) [reply contrib/babelfishpg_tsql/src/procedures.c]: Updated in latest rev.

File: contrib/babelfishpg_tsql/src/procedures.c (hunk 2 above)
Comment (ahmed-shameem) [reply contrib/babelfishpg_tsql/src/procedures.c:816]: Kept it in two separate if intentionally as this is a helper function hence it can be called by some other function. Then when the outer if satisfies, we do not know what will be the behaviour at that time. So, better to put extra check for sp_describe_undeclared_parameters.

File: contrib/babelfishpg_tsql/src/procedures.c (hunk 3 above)
Comment (ahmed-shameem) [reply contrib/babelfishpg_tsql/src/procedures.c]: Updated the unnecessary change in latest rev
This is a different error and expected. This will be encountered when user tries to perform sp_descirbe_undeclared_parameters on some non-existent column.

File: contrib/babelfishpg_tsql/src/procedures.c (hunk 4 above)
Comment (ahmed-shameem) [reply contrib/babelfishpg_tsql/src/procedures.c:721]: Kept it in two separate if intentionally as this is a helper function hence it can be called by some other function. Then when the outer if satisfies, we do not know what will be the behaviour at that time. So, better to put extra check for sp_describe_undeclared_parameters.

File: contrib/babelfishpg_tsql/src/procedures.c (hunk 5 above)
Comment (ahmed-shameem) [reply contrib/babelfishpg_tsql/src/procedures.c]: Updated the unnecessary change in latest rev
This is a different error and expected. This will be encountered when user tries to perform sp_descirbe_undeclared_parameters on some non-existent column.

**Review by timchang514 — APPROVED** [review]

**Review by shardgupta — APPROVED** [review]
